src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -2778,8 +2779,8 @@ void PhaseIdealLoop::do_range_check(IdealLoopTree *loop, Node_List &old_new) {
   // new pre_limit can push Bool/Cmp/Opaque nodes down (when one of the eliminated condition has parameters that are not
   // loop invariant in the pre loop.
   set_ctrl(pre_opaq, new_limit_ctrl);
```

Comment:
Can you update this comment to explain different control settings as you did in PR's description.